src/main/java/org/jenkinsci/plugins/vsphere/builders/Deploy.java

```diff
@@ -116,6 +170,16 @@ private boolean deployFromTemplate(final AbstractBuild<?, ?> build, final Launch
 
         vsphere.deployVm(expandedClone, expandedTemplate, linkedClone, resourcePoolName, expandedCluster, expandedDatastore, powerOn, jLogger);
 		VSphereLogger.vsLogger(jLogger, "\""+expandedClone+"\" successfully deployed!");
+		IP = vsphere.getIp(vsphere.getVmByName(expandedClone), 20);
```

Comment:
Was not enough. Should it be optionally configurable in VSphere? (Cloning step might use it as well)
It might be valuable to have this work on powerOn, similarly to what Cloning does. Alternatively - configurable.
Got: java.lang.IllegalArgumentException: Null value not allowed as an environment variable: VSPHERE_IP